### ISSUE TITLE
[NO ISSUE] "Library was NOT tested with IIS" -> It works !

### ISSUE THREAD

**Issue by TomyCesaille** [rdvojmoc/DinkToPdf#50]
Hello,
I noticed the "Library was NOT tested with IIS." in README.

For the record, the package works under IIS.
* Package 1.0.8 with libwkhtmltox.dll v0.12.4 
* IIS Version 8.5.9600.16384
* Project is almost a raw ASP.NET CORE 2.1 API template, targeting core
* Published as self contained
* I had to use the CustomAssemblyLoadContext issue resolution [here](https://github.com/rdvojmoc/DinkToPdf/issues/5#issuecomment-321183899)

Regards,

**Comment by lkolodziey** [comment]
Yes, I did a test right now and works fine.

**Comment by codingbadger** [comment]
What did you do to get this working? I am not having any luck getting this running in IIS. I get the error `Qt: Could not initialize OLE (error 80010106)`

**Comment by lkolodziey** [comment]
@codingbadger, did you copied the libwkhtmltox.dll to the folder containing your solution dlls?
I did that mistake (I didn´t copy it) and the same error appears to me.

**Comment by codingbadger** [comment]
Hi @lkolodziey yes I have done that. It is loading the dll correctly because if I put the 64bit version in, the app crashes with a Bad Image Exception.

**Comment by lkolodziey** [comment]
@codingbadger, try to put all related files like attached image and try again.
![capture](https://user-images.githubusercontent.com/6719750/45369970-9353f700-b5bd-11e8-8442-d06cbe443f45.PNG)
Let´s testing :D Does not matter if you using windows put all files ...

**Comment by codingbadger** [comment]
Yep - I have all those files :)

**Comment by lkolodziey** [comment]
It is so strange.
OK, Is your application running in IIS with "no managed code" (application pool) ?
I your application 64bit configured in IIS (application pool) ?

**Comment by codingbadger** [comment]
Yes - App pool is running as "no managed code". The "enable 32 bit applications" is set to False

**Comment by codingbadger** [comment]
I finally got to the bottom of this, it turns out the the absolute url being used for images, css etc wasn't accessible from the webserver. Once this was fixed the PDF's were generated fine.

Thanks for your help @lkolodziey

**Comment by lkolodziey** [comment]
Basically broken links, good thing that solved.
You are welcome.

**Comment by lkolodziey** [comment]
For all ... I created an example of use: https://github.com/lkolodziey/UsingDinkToPdf_Example

**Comment by novidaws** [comment]
Hello, @lkolodziey . Thank you for your sharing, I need your help, because libwkthmltox.dll is not running well in my IIS environment. 
I use libwkhtmltox.dll version 0.12.4 for .Net Core 2.0, and I published to IIS version 10.0.14393.0.

I got something error in my browser:

`An error occurred during the compilation of a resource required to process this request. Please review the following specific error details and modify your source code appropriately.
C:\inetpub\wwwroot\prisma\Views\Home\Index.cshtml

Metadata file 'C:\inetpub\wwwroot\prisma\libwkhtmltox.dll' could not be opened -- PE image doesn't contain managed metadata.
@{`

Could you share your knowledge about this?
I really appreciate your help! Thanks!

**Comment by rajputranjeet569** [comment]
Hi All,

Can i use DinkToPdf with dot net core 2.1?.

I have used it with Asp.net core. it's working on local IIS but not working on Live Server.

Error : Unable to load DLL 'C:\inetpub\wwwroot\myproject\api\bin\libwkhtmltox.dll' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E)
Unable to load DLL 'C:\inetpub\wwwroot\myproject\api\bin\libwkhtmltox.dll' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E).

Please let me know ASAP.

Thanks & Regards,
Ranjeet

**Comment by casaulenbo** [comment]
Yes you can. Performance of IIS is very bad. Imagine, at first server got 100 request concurrently, which every request will create 1 pdf file. If there is request 101th,  it have to wait until 100 request finish.

**Comment by raissahonorato** [comment]
Hello, I'm from Brazil and I saw that this post is a little bit like my problem.
my application returns the following result:
"A#Could not load file or assembly 'DinkToPdf, Version=1.0.8.0, Culture=neutral, PublicKeyToken=null'"
have you seen something like that?
thanks for listening!

**Comment by casaulenbo** [comment]
Have you try this before?
https://github.com/rdvojmoc/DinkToPdf/issues/5#issuecomment-321183899

**Comment by raissahonorato** [comment]
yes, but still giving this error
do i need to add an address to this iis option?
![Sem título](https://user-images.githubusercontent.com/20980022/80217824-b4dce780-8616-11ea-975d-a30bc661c487.png)

**Comment by raissahonorato** [comment]
{
  "Data": {},
  "Result": 1,
  "Message": " Could not load file or assembly 'DinkToPdf, Version=1.0.8.0, Culture=neutral, PublicKeyToken=null'. O sistema não pode encontrar o arquivo especificado.",
  "DebugMessage": ""
}

**Comment by arpitce27** [comment]
> @codingbadger, did you copied the libwkhtmltox.dll to the folder containing your solution dlls? I did that mistake (I didn´t copy it) and the same error appears to me.

And as a part of problem I did installed below, and I tried removing and running application - it throw same error. after re installation it worked fine.
https://learn.microsoft.com/en-US/cpp/windows/latest-supported-vc-redist?view=msvc-170

This was the key for my .NET CORE 2.2 Project.

Thank you @lkolodziey
